website/templates/part.html

```diff
@@ -123,9 +123,82 @@ <h5 class="mb-4">Author: <a href ="{{ url_for('views.userView', user_name=author
+                                    <small class="text-muted"><time datetime="{{ comment.date.isoformat(timespec='seconds') }}">{{ comment.date }}</time></small>
+                                </div>
+                                <p class="mb-1">{{ comment.content }}</p>
+                            </div>
+                            {% if current_user.is_authenticated %}
+                            <button class="btn btn-sm btn-link reply-btn ps-0" data-comment-id="{{ comment.id }}">Reply</button>
+                            <form method="POST" action="{{ url_for('views.part', part_number=part.id) }}" class="reply-form mt-2 d-none needs-validation" novalidate>
+                                <input type="hidden" name="parent_id" value="{{ comment.id }}">
+                                <textarea name="content" class="form-control form-control-sm" rows="2" required></textarea>
+                                <button type="submit" class="btn btn-sm btn-primary mt-1">Reply</button>
+                                <input type="hidden" name="_csrf_token" value="{{ csrf_token() }}">
+                            </form>
+                            {% endif %}
+
+                            {% if comment.replies %}
+                            <div class="replies mt-3">
+                                {% for reply in comment.replies %}
+                                <div class="reply mb-3 ps-4 border-start">
+                                    <div class="bg-light p-3 rounded">
+                                        <div class="d-flex justify-content-between align-items-center mb-2">
+                                            <strong><a href="{{ url_for('views.userView', user_name=reply.author.username) }}" class="text-decoration-none">{{ reply.author.username }}</a></strong>
+                                            <small class="text-muted"><time datetime="{{ reply.date.isoformat(timespec='seconds') }}">{{ reply.date }}</time></small>
+                                        </div>
+                                        <p class="mb-0">{{ reply.content }}</p>
+                                    </div>
+                                </div>
+                                {% endfor %}
+                            </div>
+                            {% endif %}
+                        </div>
+                    </div>
+                </div>
+                {% else %}
+                <p class="text-muted">No comments yet.</p>
+                {% endfor %}
             </div>
         </div>
     </div>
+    </div>
+    </div>
+    </div>
 </section>
+
 <script src="{{ url_for('static', filename='js/time-formatter.js') }}" type="module" nonce="{{ csp_nonce() }}"></script>
+<script nonce="{{ csp_nonce() }}">
+document.querySelectorAll('.reply-btn').forEach(btn => {
+    btn.addEventListener('click', () => {
+        const form = btn.nextElementSibling;
+        form.classList.toggle('d-none');
+    });
+});
+</script>
```

Comment:
I'd prefer to stick with scripts in separate files rather than inline.
Also, I'm not sure if we should have a (hidden) `<form>` under every top-level comment. Maybe we could just have one that is initialised with relevant parent id and parent comment's author like in Wordpress?